src/object/object_primitive.c

```diff
@@ -15700,3 +16005,104 @@ mr_cmpval_enumeration (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, in
 
   return MR_CMP (s1, s2);
 }
+
+/*
+ * mr_get_compressed_data_from_buffer ()	  - Uncompresses a data stored in buffer.
+ *
+ * return()					  : NO_ERROR or error code.
+ * buf(in)					  : The buffer from which is needed decompression.
+ * data(out)					  : The result of the decompression. !!! Needs to be alloc'ed !!!
+ * compressed_size(in)				  : The compressed data size.
+ * decompressed_size(in)			  : The uncompressed data size.
+ */
+int
+mr_get_compressed_data_from_buffer (OR_BUF * buf, char *data, int compressed_size, int decompressed_size)
+{
+  int rc = NO_ERROR;
+
+  /* Check if the string needs decompression */
+  if (compressed_size > 0)
+    {
+      lzo_uint decompression_size = 0;
+      /* Handle decompression */
+
+      /* decompressing the string */
+      rc = lzo1x_decompress ((lzo_bytep) buf->ptr, (lzo_uint) compressed_size, data, &decompression_size, NULL);
+      if (rc != LZO_E_OK)
+	{
+	  return rc;
```

Comment:
@lucianlazar2211 

Please set `ER_IO_LZO_DECOMPRESS_FAIL` error and return the error code instead of `rc`.